launcher/ui/connectpage.cpp

```diff
@@ -55,9 +70,178 @@ ConnectPage::ConnectPage(QWidget *parent)
     connect(ui->instanceView, SIGNAL(activated(QModelIndex)), SIGNAL(activate()));
 
     QSettings settings;
-    ui->host->setText(settings.value(QStringLiteral("Connect/Host"), QString()).toString());
-    ui->port->setValue(settings.value(QStringLiteral("Connect/Port"),
-                                      Endpoint::defaultPort()).toInt());
+    ui->host->setText(settings.value(QStringLiteral("Connect/Url"), QString()).toString());
+}
+
+void ConnectPage::validateHostAddress(const QString &address)
+{
+    QString stillToParse = address;
+    m_valid = false;
+    bool correctSoFar = false;
+    m_currentUrl.clear();
+
+    // Initially, set the text to red
+    QPalette palette;
+    palette.setColor(QPalette::Text,Qt::red);
+    ui->host->setPalette(palette);
+    clearWarnings();
+
+    handleLocalAddress(stillToParse, correctSoFar);
+    handleIPAddress(stillToParse, correctSoFar);
+
+    QRegExp hostNameFormat("^([a-z,A-Z][a-z,A-Z,0-9,\\-\\.]+[a-z,A-Z,0-9])$");
+    if (hostNameFormat.exactMatch(stillToParse))
+        handleHostName(stillToParse);
+
+    // if we came down here and theres nothing more to parse, we are good
+    // set text back to black again
+    if (correctSoFar && stillToParse.isEmpty()) {
+        m_valid = true;
+        palette.setColor(QPalette::Text,Qt::black);
```

Comment:
Hardcoded black color looks problematic for dark color schemes. Can we use the text color from the system palette instead?